pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -477,10 +516,64 @@ stdenv.mkDerivation ({
     # TODO: fetch the self from the fixpoint instead
     haddockDir = self: if doHaddock then "${docdir self.doc}/html" else null;
 
-    env = shellFor {
-      packages = p: [ drv ];
-      inherit shellHook;
-    };
+    #  Creates a derivation containing all of the necessary dependencies for building the
+    #  parent derivation. The attribute set that it takes as input can be viewed as:
+    #
+    #     { withHoogle }
+    #
+    #  The derivation that it builds contains no outpaths because it is meant for use
+    #  as an environment
+    #
+    #    # Example use
+    #    # Creates a shell with all of the dependencies required to build the "hello" package,
+    #    # and with python:
+    #
+    #    > nix-shell -E 'with (import <nixpkgs> {}); \ 
+    #    >    haskell.packages.ghc865.hello.envFunc { buildInputs = [ python ]; }'
+    envFunc = { withHoogle ? false }:
```

Comment:
Does envFunc need to be exposed in in passthru? Or perhaps we can just move it to a let binding.

Reply:
I'm also unsure why env can't just use shellFor here?